src/connector/postgres/conversion.rs

```diff
@@ -221,10 +221,35 @@ impl GetRow for PostgresRow {
                     None => ParameterizedValue::Null,
                 },
                 ref x => match x.kind() {
-                    Kind::Enum(_) => {
-                        let val: EnumString = row.try_get(i)?;
-                        ParameterizedValue::Enum(val.value.into())
-                    }
+                    Kind::Enum(_) => match row.try_get(i)? {
+                        Some(val) => {
+                            let val: EnumString = val;
+                            ParameterizedValue::Enum(val.value.into())
+                        }
+                        None => ParameterizedValue::Null,
+                    },
+                    Kind::Array(inner) => match inner.kind() {
+                        Kind::Enum(_) => match row.try_get(i)? {
+                            Some(val) => {
+                                let val: Vec<EnumString> = val;
+                                ParameterizedValue::Array(
+                                    val.into_iter()
+                                        .map(|x| ParameterizedValue::Enum(x.value.into()))
+                                        .collect(),
+                                )
+                            }
+                            None => ParameterizedValue::Null,
+                        },
+                        _ => match row.try_get(i)? {
+                            Some(val) => {
+                                let val: Vec<String> = val;
+                                ParameterizedValue::Array(
+                                    val.into_iter().map(|x| ParameterizedValue::Text(x.into())).collect(),
```

Comment:
Hmm, looking at this aren't we just converting everything that is not an enum to a string? What about integers? This smells like we could just call the function recursively with the inner kind to get a proper conversion...

Reply:
Yes, we could probably do that. But I do not want to do this now since I only need enum arrays. This just handles enums and the rest is handled like before.